hudi-common/src/main/java/org/apache/hudi/metadata/AbstractHoodieTableMetadata.java

```diff
@@ -35,16 +36,18 @@
 public abstract class AbstractHoodieTableMetadata implements HoodieTableMetadata {
 
   protected transient HoodieEngineContext engineContext;
+  protected transient HoodieStorage storage;
```

Comment:
Trying to understand what necessitated this change? I mean why do we need `HoodieStorage` in addition to `StorageConfiguration`?

Reply:
`storageConf` is serializable so that instantiating the `storage` instance (not serializable) on the executor side uses the `storageConf`.